src/devices/MotorHat/MotorHat.cs

```diff
@@ -0,0 +1,193 @@
+            }
+
+            //Add to the channelsUsed list, the channels we will use for this DCMotor
+            var speedPwm = pca9685.CreatePwmChannel(speedPin);
+            var in1Pwm = pca9685.CreatePwmChannel(in1Pin);
+            var in2Pwm = pca9685.CreatePwmChannel(in2Pin);
+
+            channelsUsed.Add(speedPwm);
+            channelsUsed.Add(in1Pwm);
+            channelsUsed.Add(in2Pwm);
+
+            var dcMotor = new DCMotor3Pwm(speedPwm, in1Pwm, in2Pwm);
+
+            return dcMotor;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="PwmChannel"/> from one of the 4 available PWM channels in the MotorHat
+        /// </summary>
+        /// <param name="channelNumber">A valid PWM channel (0, 1, 14 or 15)</param>
+        /// <remarks>
+        /// The channelNumber refers to ont of the available PWM channel numbers available in the Motor Hat (0, 1, 14, 15) printed in the device.
+        /// </remarks>
+        public PwmChannel CreatePwmChannel(byte channelNumber)
+        {
+            if (channelNumber != 0 && channelNumber != 1 && channelNumber != 14 && channelNumber != 15)
+            {
+                throw new ArgumentOutOfRangeException(nameof(channelNumber), $"Must be one of de available PWM channels (0, 1, 14 or 15). (received: {channelNumber})");
+            }
+
+            var pwmChannel = this.pca9685.CreatePwmChannel(channelNumber);
+
+            channelsUsed.Add(pwmChannel);
+
+            return pwmChannel;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ServoMotor"/> from one of the 4 available PWM channels  in the MotorHat
+        /// </summary>
+        /// <param name="channelNumber">A valid PWM channel (0, 1, 14 or 15)</param>
+        /// <param name="maximumAngle">The maximum angle the servo motor can move represented as a value between 0 and 360.</param>
+        /// <param name="minimumPulseWidthMicroseconds">The minimum pulse width, in microseconds, that represent an angle for 0 degrees.</param>
+        /// <param name="maximumPulseWidthMicroseconds">The maxnimum pulse width, in microseconds, that represent an angle for maximum angle.</param>
+        /// <remarks>
+        /// The channelNumber refers to ont of the available PWM channel numbers available in the Motor Hat (0, 1, 14, 15) printed in the device.
+        /// </remarks>
+        public ServoMotor.ServoMotor CreateServoMotor(byte channelNumber, double maximumAngle = 180, double minimumPulseWidthMicroseconds = 1000, double maximumPulseWidthMicroseconds = 2000)
+        {
+            var pwmChannel = this.CreatePwmChannel(channelNumber);
+
+            var servo = new ServoMotor.ServoMotor(pwmChannel, maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds);
+
+            return servo;
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            foreach (var channel in channelsUsed)
```

Comment:
AFAIR pca9685 doesn't require disposing of the channels - the channels Dispose don't really do anything useful (it only marks it as disposed but that's only used for tracking on CreateChannel so that you don't accidentally create two channels). Either way, it should be Pca9685's responsibility to clean up since channels are not usable after it is disposed.

Reply:
(also you can test if stopping is needed - I can't remember from the top of my head)

Reply:
I'll remove the channel disposing then.

Stopping the motors is not strictly required, but if I don't, then they keep spinning after the app exited.
I am planning to use this for a dotnet based lawn mowing robot.. If motors remain spinning someone could lose a finger :O. I could handle this outside of the MotorHat device binding code though if you think that would be better.

Reply:
Yes, I normally try not to do anything in the dispose method of other bindings but this one seems fine for the reason above